### ISSUE TITLE
[Patch] IrDA capture support

### ISSUE THREAD

**Issue by guyharris** [the-tcpdump-group/libpcap#205]
Converted from [SourceForge issue 836210](http://sourceforge.net/support/tracker.php?aid=836210), submitted by jean2

```
    Hi,

    This patch adds support for IrDA packet capture
```

to libpcap v0.7.1.

```
    Why ? IrDA is not Ethernet and not TCP/IP, so
```

why bother with IrDA support ?
        In some cases, IP frames may be encapsulated in
IrDA sockets (IrLAN, IrCOMM+PPP, IrNET), and therefore
enabling IrDA support in tcpdump would allow full
decoding of IrDA _and_ TCP/IP headers, which is not
possible with today's tools. Of course, before we can
add that in tcpdump, we need libpcap support.

```
    Why a cooked socket ?
    The decoding of IrDA frames depend on the
```

direction of the frame (master or slave), and this is
not part of the frame. Therefore, a pseudo header need
to be added to each frame, containing the frame
direction. Rather than invent yet another
pseudo-header, or to subverting the pseudo-header of
another protocol, I'd rather reuse the SLL header.
        Why the new DL_IRDA, why not use DL_LINUX_SLL ?
        Libpcap assume that sockets of type
DL_LINUX_SLL contains IP frames and allows to create
TCP/IP filter on those. On the other hand, an IrDA
socket doesn't contain IP frames, and you should not be
able to set TCP/IP filter on those.
        Does using the linux_sll pseudo header make it
Linux specific ?
        Not at all, it just means a bit more work for
other platforms, but they would have to add a pseudo
header anyway. Anyway, most of those other platforms
don't have an IrDA stack...

```
    Please send me useful feedback so that we can
```

get this patch properly integrated.
        Thanks,

```
    Jean
```
